azurelinuxagent/ga/monitor.py

```diff
@@ -418,10 +418,36 @@ def send_cgroup_telemetry(self):
 
         if datetime.datetime.utcnow() >= (self.last_telemetry_heartbeat + MonitorHandler.CGROUP_TELEMETRY_PERIOD):
             try:
-                for cgroup_name, metrics in CGroupsTelemetry.collect_all_tracked().items():
+                metric_reported, metric_threshold = CGroupsTelemetry.collect_all_tracked()
+                for cgroup_name, metrics in metric_reported.items():
+                    thresholds = metric_threshold[cgroup_name]
+
                     for metric_group, metric_name, value in metrics:
                         if value > 0:
                             report_metric(metric_group, metric_name, cgroup_name, value)
+
+                        if metric_group == "Memory":
+                            if value > thresholds["memory"]:
+                                msg = "CGroup {0}: Crossed the Memory Threshold. Current Value:{1}, Threshold:{2}.".format(
+                                    cgroup_name, value, thresholds["memory"])
+                                add_event(name=AGENT_NAME,
+                                          version=CURRENT_VERSION,
+                                          op=WALAEventOperation.CGroupsLimitsCrossed,
+                                          is_success=True,
+                                          message=msg,
+                                          log_event=True)
+
+                        if metric_group == "Process":
+                            if value > thresholds["cpu"]:
```

Comment:
```suggestion
                            if value >= thresholds["cpu"]:
```

Cgroups should throttle cpu consumption to not exceed the defined limit, which means it might equal the limit. You still want telemetry to report "this agent/extension hit the defined limit".